core/shared/src/main/scala/caseapp/core/util/Formatter.scala

```diff
@@ -0,0 +1,23 @@
+package caseapp.core.util
+
+import caseapp.Name
+
+abstract class Formatter[T] {
+  def format(t: T): String
+}
+
+object Formatter {
+
+  /**
+    * Formats option arguments to a given format.
+    *
+    * Default formatter will format option arguments as `foo-bar`.
+    */
+  val DefaultNameFormatter = new Formatter[Name] {
```

Comment:
Let's explicitly type this as `Formatter[Name]`, so that the structural type doesn't leak in the public API.
```suggestion
  val DefaultNameFormatter: Formatter[Name] = new Formatter[Name] {
```

Reply:
I'm always in favor of explicitly adding the type 😄